### DISCUSSION TITLE
lego CLI equivalent to `certbot` `--manual-auth-hook` and `--manual-cleanup-hook`

### DISCUSSION THREAD

**Discussion by abdeldjalil-fellah** [go-acme/lego#2331] [Q&A] [answered]
Does lego CLI have an equivalent to `certbot` `--manual-auth-hook` and `--manual-cleanup-hook` to be used with custom DNS provider instead of using predefined DNS providers and library challenge.Provider?

**Comment by ldez** [comment, marked as answer]
Hello,

Yes, you can use one of these possibilities:
- https://go-acme.github.io/lego/dns/manual/index.html
- https://go-acme.github.io/lego/dns/exec/index.html
- https://go-acme.github.io/lego/dns/httpreq/index.html

**Comment by abdeldjalil-fellah** [comment]
What about putting `manual`, `exec` and `httpreq` in a separate category named "Custom DNS Providers" and the rest in "Popular DNS Providers"?

**Reply by ldez** [reply]
What is your DNS provider?

**Reply by abdeldjalil-fellah** [reply]
Technitium

**Reply by ldez** [reply]
Can you share a script that you are currently using?

I have not yet read the [API documentation](https://github.com/TechnitiumSoftware/DnsServer/blob/master/APIDOCS.md), but I think I can create a dedicated implementation if you are interested. I will need you to test the implementation.

**Reply by abdeldjalil-fellah** [reply]
I did not write a script yet

**Reply by ldez** [reply]
If I create an implementation, can you test the PR?

I will explain to you everything you need to test the PR if you agree.

**Reply by abdeldjalil-fellah** [reply]
Not sure, but I will do my best.

**Reply by ldez** [reply]
Why are you not sure?
Because you don't know the test process or because of something else?

I need to know because my time is limited so I don't want to spend time on something if nobody will test it.

**Reply by abdeldjalil-fellah** [reply]
I’m currently focused on some important projects that require my full attention. As a result, I won’t be able to test until I complete these projects, which may take more than a month.

**Reply by ldez** [reply]
I created an implementation: https://github.com/go-acme/lego/pull/2332


I also found this article: https://blog.technitium.com/2023/03/

However, Technitium is compatible with the [RFC2136 provider](https://go-acme.github.io/lego/dns/rfc2136/index.html), as the article describes.
